kgfx.cpp

```diff
@@ -62,6 +62,24 @@ void KGFX::drawText(TFT_eSprite &spr, const char *txt, const tftfont_t &f, int c
   spr.pushSprite(x, y);
 }
 
+/***************************************************************************************
+ * Function name:           drawTextCentered
+ * Description:             Draws text to given sprite centered
+ ***************************************************************************************/
+void KGFX::drawTextCentered(TFT_eSprite &spr, const char *txt, const tftfont_t &f, int color, int y)
+{
+  tft.TTFdestination(&spr);
+  // spr.fillSprite(TFT_BLACK); // Do not fill sprite with black, it will overwrite the sprite
```

Comment:
I think we need this in order to refresh the sprite to show the new text.

Reply:
In my tests, if this line is uncommented, each subsequent line of text erases the previous one.

Reply:
hmm weird, it doesn't erase the previous text for me. What text were you displaying in sequence? I toggled the following:

```
  if (txt == "hello") {
    txt = "hi";
  } else {
    txt = "hello";
  }
  ui.drawTextCentered(spr, txt, Arial_16, TFT_SKYBLUE, 0);
```

It needed fillSprite in order to clear the previous text. Can you see if this pair of text works at your end?

Reply:
The following code will print the first line, then erase it and print the second, then erase that and print the third. 
```c
if(INFLUXDB_URL == "" || INFLUXDB_TOKEN == "" || INFLUXDB_ORG == "" || INFLUXDB_BUCKET == "" || INFLUXDB_QUERY == "" || STAT_LABEL == "" || STAT_TYPE == "") {
    ui.drawTextCentered(textSpr, "Please set", Arial_28, TFT_RED, 2);
    ui.drawTextCentered(textSpr, "InfluxDB", Arial_28, TFT_RED, 34);
    ui.drawTextCentered(textSpr, "Credentials", Arial_28, TFT_RED, 66);
    while(1) {
      delay(1000);
      if ((WiFi.status() == WL_CONNECTED)) { otaserver.handle(); }
    }
  }
```

Reply:
What if you set the last param to 0 for all 3 lines?

It erases the previous text probably because the sprite is now in a different location. However, if the sprite doesn't move, the previous text isn't cleared.

Reply:
Oh, I see the issue now.

We want to keep the fillSprite as it allows the text to be overwritten by the next one.

To display all 3 lines, can you try the following?

Option 1: Create 3 separate sprites:
```
TFT_eSprite spr1 = ui.createSprite(240, 30);
ui.drawTextCentered(spr1, "Please set", Arial_28, TFT_RED, 2);
TFT_eSprite spr2 = ui.createSprite(240, 30);
ui.drawTextCentered(spr2, "Credentials", Arial_28, TFT_RED, 34);
...
```

Option 2: Use drawText without sprite if you don't want the values to be erased.

Reply:
> What if you set the last param to 0 for all 3 lines?
> 

Then it just over-writes the first line with each subsequent line. Not at all what I want. The last param is the `y` coordinate, so setting it to 0 all the time isn't an option if you want more than one line of text. 

The second option is I guess viable, but it is a massive waste of memory as you have to have a separate sprite for each line of text. 

Here are the tests I just ran: 
1. Using just `ui.drawText()`
2. using `ui.drawTextCentered()` with no call to filling the sprite with black. 
3. using `ui.drawTextcentered()` filling the screen with black
4. using `ui.drawTextCentered()` filling the sprite with black.

https://github.com/user-attachments/assets/db99fd43-7071-4b1a-b5fb-e4714cbab508

https://github.com/user-attachments/assets/a2959d70-f384-4f1b-92a5-c10f528ae96c

https://github.com/user-attachments/assets/d6108924-6358-4561-a7f8-3fb74e7293fa

https://github.com/user-attachments/assets/466c4fa8-72e2-47e7-9ddd-05ef31d8de62

Reply:
Is the text meant to be refreshed? Or is it static text?

Reply:
In this instance, static text.

Reply:
There are 2 functions for drawText, one takes in a sprite param, the other does not. The one with the sprite param enables writing a new text on the same sprite without bothering about erasing the previous one. For static text the simple drawText function should work since we don't need the refresh feature. 

So any function that takes in the sprite should support the refresh by including the fillSprite line. Is there a reason why you needed to use the sprite?